packages/protocol/test/common/onboarding.ts

```diff
@@ -0,0 +1,424 @@
+import {
+  assertEqualBN,
+  assumeOwnership,
+  getDerivedKey,
+  getVerificationCodeSignature,
+  KeyOffsets,
+  unlockAndAuthorizeKey,
+} from '@celo/protocol/lib/test-utils'
+import { getDeployedProxiedContract } from '@celo/protocol/lib/web3-utils'
+import { ensureLeading0x, privateKeyToAddress, trimLeading0x } from '@celo/utils/lib/address'
+import {
+  AccountsContract,
+  AccountsInstance,
+  AttestationsTestContract,
+  AttestationsTestInstance,
+  MetaTransactionWalletContract,
+  MetaTransactionWalletDeployerContract,
+  MetaTransactionWalletDeployerInstance,
+  MetaTransactionWalletInstance,
+  MockRandomContract,
+  MockRandomInstance,
+  MockValidatorsContract,
+  ProxyCloneFactoryContract,
+  ProxyCloneFactoryInstance,
+  ProxyContract,
+  ProxyInstance,
+  RegistryInstance,
+  StableTokenInstance,
+} from 'types'
+
+const executeMetaTransaction = async (
+  signer: string,
+  mtw: MetaTransactionWalletInstance,
+  metaTx: MetaTransaction
+) => {
+  const { v, r, s } = await getSignatureForMetaTransaction(signer, mtw.address, metaTx)
+  return mtw.executeMetaTransaction(metaTx.destination, metaTx.value, metaTx.data, v, r, s)
+}
+
+const getExecuteMetaTransactionData = async (
+  signer: string,
+  mtw: MetaTransactionWalletInstance,
+  metaTx: MetaTransaction
+) => {
+  const { v, r, s } = await getSignatureForMetaTransaction(signer, mtw.address, metaTx)
+  // @ts-ignore
+  return mtw.contract.methods
+    .executeMetaTransaction(metaTx.destination, metaTx.value, metaTx.data, v, r, s)
+    .encodeABI()
+}
+
+const logCosts = false
+const logCost = (output: string) => {
+  if (logCosts) {
+    // tslint:disable-next-line:no-console
+    console.log(output)
+  }
+}
+
+const Accounts: AccountsContract = artifacts.require('Accounts')
+const Attestations: AttestationsTestContract = artifacts.require('AttestationsTest')
+const MTW: MetaTransactionWalletContract = artifacts.require('MetaTransactionWallet')
+const Random: MockRandomContract = artifacts.require('MockRandom')
+contract('Komenci Onboarding', (_accounts: string[]) => {
+  let accounts: AccountsInstance
+  let stableToken: StableTokenInstance
+  let mtw: MetaTransactionWalletInstance
+  let random: MockRandomInstance
+  let registry: RegistryInstance
+  let relayermtw: MetaTransactionWalletInstance
+  let transactions: any[]
+  const attestationExpiryBlocks = (60 * 60) / 5
+  const selectIssuersWaitBlocks = 4
+  const maxAttestations = 20
+  const user: string = _accounts[1]
+  // Random hex strings
+  const dek = '0x02f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e01611111111'
+  const identifier = '0x02f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e016111111'
+  const numAttestations = 3
+  // TODO(asa): Clean this up
+  // const perAttestationFee = web3.utils.toWei(numAttestations * 0.05, 'ether').toString()
+  const perAttestationFee = '50000000000000000'
+  const attestationFee = '150000000000000000'
+  before(async () => {
```

Comment:
Should this be a `beforeEach` to ensure a clean environment before each test? Ditto for other `before`s in this file.

Reply:
What is there in here that you would want to repeat per-test? My intention was to put the one time setup in this, and the per-test setup in their respective `before`s. Note that because we are using the contracts deployed by the migrations you don't get a clean environment no matter what you do.

Reply:
Should we use mocks here instead of rely on the migrations contracts?

Reply:
I would prefer not to since this is intended to be an integration test